From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Victor Fuentes <victor@vlinkz.dev>
Date: Wed, 29 Jul 2026 00:24:47 -0700
Subject: [PATCH] power: sequencing: qcom-wcn: Add support for WCN7860

Compared to WCN7850 it takes VDD_CX and the antenna supply instead of
VDD and VDD_IO.

Signed-off-by: Victor Fuentes <victor@vlinkz.dev>
---
 drivers/power/sequencing/pwrseq-qcom-wcn.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/power/sequencing/pwrseq-qcom-wcn.c b/drivers/power/sequencing/pwrseq-qcom-wcn.c
index 636dd7e63891..31d7ac32b54f 100644
--- a/drivers/power/sequencing/pwrseq-qcom-wcn.c
+++ b/drivers/power/sequencing/pwrseq-qcom-wcn.c
@@ -431,6 +431,23 @@ static const struct pwrseq_qcom_wcn_pdata pwrseq_wcn7850_of_data = {
 	.targets = pwrseq_qcom_wcn_targets,
 };
 
+static const char *const pwrseq_wcn7860_vregs[] = {
+	"vddio1p2",
+	"vddant",
+	"vddaon",
+	"vdddig",
+	"vddrfa1p8",
+	"vddrfa1p2",
+	"vddcx",
+};
+
+static const struct pwrseq_qcom_wcn_pdata pwrseq_wcn7860_of_data = {
+	.vregs = pwrseq_wcn7860_vregs,
+	.num_vregs = ARRAY_SIZE(pwrseq_wcn7860_vregs),
+	.pwup_delay_ms = 50,
+	.targets = pwrseq_qcom_wcn_targets,
+};
+
 static int pwrseq_qcom_wcn_match_regulator(struct pwrseq_device *pwrseq,
 					   struct device *dev,
 					   const char *name)
@@ -592,6 +609,10 @@ static const struct of_device_id pwrseq_qcom_wcn_of_match[] = {
 		.compatible = "qcom,wcn7850-pmu",
 		.data = &pwrseq_wcn7850_of_data,
 	},
+	{
+		.compatible = "qcom,wcn7860-pmu",
+		.data = &pwrseq_wcn7860_of_data,
+	},
 	{
 		.compatible = "qcom,wcn6750-pmu",
 		.data = &pwrseq_wcn6750_of_data,
